packages/cache/src/internal/requestUtils.ts

```diff
@@ -111,7 +111,9 @@ export async function retryTypedResponse<T>(
         return {
           statusCode: error.statusCode,
           result: null,
-          headers: {}
+          headers: {},
+          message: error.message,
+          typeKey: error.result.typeKey
```

Comment:
Instead of adding these two separately can we add `error` directly. I think the type is [HttpClientError](https://github.com/actions/http-client/blob/5ea1c5292929dba26848ea46f3b4c2e1ea1e30a2/index.ts#L72)

Reply:
Added `error: HttpClientError` instead of these two fields.